service/src/main/java/bio/terra/cbas/service/RunSetsService.java

```diff
@@ -357,6 +378,25 @@ private List<RunStateResponse> buildInputsAndSubmitRun(
             cromwellService.submitWorkflowBatch(
                 rawMethodUrl, engineIdToWorkflowInput, workflowOptionsJson, userToken);
 
+        if (batchIdx == 0) {
+          micrometerMetrics.stopTimer(
+              cromwellInitialRequestTimerSample,
+              "cromwell_request_to_initial_submission_timer",
+              "run_set_id",
+              runSet.runSetId().toString(),
+              "requestSuccessful",
+              "true");
+        }
+        if (batchIdx == batches.size() - 1) {
```

Comment:
small nit: it would be good to add comments to describe what these metrics here are recording